deployment/src/main/java/io/quarkiverse/jackson/jq/deployment/JacksonJqProcessor.java

```diff
@@ -1,59 +1,96 @@
-import io.quarkus.deployment.builditem.nativeimage.NativeImageResourcePatternsBuildItem;
-import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
 import io.quarkus.deployment.recording.RecorderContext;
+import io.quarkus.runtime.RuntimeValue;
+import net.thisptr.jackson.jq.BuiltinFunction;
 import net.thisptr.jackson.jq.BuiltinFunctionLoader;
-import net.thisptr.jackson.jq.Function;
 import net.thisptr.jackson.jq.Scope;
-import net.thisptr.jackson.jq.Versions;
 
 class JacksonJqProcessor {
-
     private static final String FEATURE = "jackson-jq";
-    private static final String REFLECTION_VERSION_RANGE_DESERIALIZER = "net.thisptr.jackson.jq.internal.misc.VersionRangeDeserializer";
-    private static final String JSON_CONFIG_GLOB = "net/thisptr/jackson/**/*.json";
 
     @BuildStep
     FeatureBuildItem feature() {
         return new FeatureBuildItem(FEATURE);
     }
 
     @BuildStep
-    List<ReflectiveClassBuildItem> registerForReflection() {
-        final List<ReflectiveClassBuildItem> reflectionConfig = new ArrayList<>();
-        reflectionConfig.add(ReflectiveClassBuildItem.weakClass(REFLECTION_VERSION_RANGE_DESERIALIZER));
-        return reflectionConfig;
-    }
+    @Record(ExecutionTime.STATIC_INIT)
+    SyntheticBeanBuildItem quarkusScopeBean(
+            CombinedIndexBuildItem combinedIndex,
+            ApplicationArchivesBuildItem archives,
+            JacksonJqConfig config,
+            JacksonJqScopeRecorder recorder,
+            RecorderContext context) throws Exception {
 
-    @BuildStep
-    NativeImageResourcePatternsBuildItem includeJsonConfigFile() {
-        return NativeImageResourcePatternsBuildItem.builder().includeGlob(JSON_CONFIG_GLOB).build();
-    }
+        IndexView indexView = combinedIndex.getIndex();
+        List<String> excludes = config.functions.excludes.orElseGet(Collections::emptyList);
+
+        RuntimeValue<Scope> root = recorder.createScope();
+        RuntimeValue<Scope> local = recorder.createScope(root);
+
+        ClassLoader cl = BuiltinFunctionLoader.class.getClassLoader();
+
+        // load build int functions
+        lookupFunctionsFromConfig(archives, config).forEach(e -> {
+            if (!excludes.contains(e.name)) {
+                recorder.addFunction(root, e.name, e.args, e.body, config.functions.version.toString());
+            }
+        });
+        lookupFunctions(indexView, config, context, BuiltinFunction.class).forEach(f -> {
+            if (!excludes.contains(StringUtils.substringBefore(f.getName(), '/'))) {
+                recorder.addFunction(local, f.getName(), f.getFunction());
+            }
+        });
+        lookupFunctions(indexView, config, context, net.thisptr.jackson.jq.internal.BuiltinFunction.class).forEach(f -> {
+            if (!excludes.contains(StringUtils.substringBefore(f.getName(), '/'))) {
+                recorder.addFunction(local, f.getName(), f.getFunction());
+            }
+        });
```

Comment:
@lburgazzoli For a security reason, `env/0` should not be loaded by default. In jackson-jq, we exclude `env/0` by not adding `@AutoService(Function.class)` annotation on the EnvFunction so that it will not be listed in the service loader manifest.

https://github.com/eiiches/jackson-jq/blob/dcb4836162edcb6a3ef9f18d3da396f0b23937d5/jackson-jq/src/main/java/net/thisptr/jackson/jq/internal/functions/EnvFunction.java#L17-L18

I think I can just remove `@BuiltinFunction("env/0")` from the `EnvFunction` in jackson-jq side, so that this Quarkus integration doesn't need to perform some kind of extra filtering (the annotation is meaningless without `@AutoService` anyway). Wdyt?

Reply:
Maybe we can also remove the `EnvFunction` from the jandex to make sure it's not loaded by default.

Reply:
I was about to merge https://github.com/eiiches/jackson-jq/pull/196, but I'll wait. The PR looks good otherwise!

Reply:
The `@BulitinFunction("env/0")` is removed in https://github.com/eiiches/jackson-jq/pull/197 .